### ISSUE TITLE
No search engine logo in Firefox for Android

### ISSUE THREAD

**Issue by nuntius35** [openstreetmap/openstreetmap-website#1145]
1. Navigate to openstreetmap.org with Firefox for Android.
2. Long tap into the search field and select the "add search engine"-button.

Expected result: A button with the OpenStreetMap logo is added as a new search provider.

Actual result: A button with the default search logo (a magnifying glass) is added as a new search provider.

This behaviour makes it annoying to use the OSM search in Firefox for Android.

**Comment by floscher** [comment]
Do you know a site that does this? Because even http://google.com and http://youtube.com don't do this. Both get the same default search logo, so I doubt, that what you want is possible without changes by the Firefox for mobile developers.

As an alternative you could search with DuckDuckGo (in Firefox Mobile by default, the duck-logo on a red circle) and add ` !osm ` to your search. This redirects you directly to openstreetmap.org with your search loaded. That's how I search OSM on mobile.
They have plenty of other interesting so called [!bang](https://duckduckgo.com/bang)s: `!osmw` for the Openstreetmap Wiki, `!otw` for the Overpass turbo wizard, `!g` for Google, `!gh` for GitHub, `!yt` for YouTube, …

**Comment by nuntius35** [comment]
For me, the logo works on most sites but for some it doesn't. I don't think it has something to do with HTTPS. However I couldn't spot any other obvious pattern when it works or not. Maybe this is really something Firefox has to fix?

Thank you for the constructive hint using DuckDuckGo bangs. I didn't know about it.

**Comment by floscher** [comment]
Today I tried it again and it worked for every site I tried (including OSM, see below). Yesterday I wasn't able to find a site where it works. So most probably this is an issue with Firefox for Android.
As far as I understand they simply take the favicon, so no special preparation of the site should be needed.
![tmp_5785-screenshot_2016-02-04-11-45-58-800170059](https://cloud.githubusercontent.com/assets/3904348/12812903/04ffae88-cb35-11e5-9e5f-7598fe69b8ed.png)
